components/buildless-serverless/internal/controller/function-validator/function_validator.go

```diff
@@ -0,0 +1,107 @@
+package function_validator
+
+import (
+	"errors"
+	"fmt"
+	serverlessv1alpha2 "github.com/kyma-project/serverless/api/v1alpha2"
+	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
+	"slices"
+	"strings"
+)
+
+type functionValidator struct {
+	instance *serverlessv1alpha2.Function
+}
+
+func NewFunctionValidator(instance *serverlessv1alpha2.Function) *functionValidator {
```

Comment:
I would suggest renaming this func to `validator` for the package and `New` for the function, then instead of having `function_validator.NewFunctionValidator` that is running this piece of code we should have `validator.New` or we can put `Function` word after `New` to got something like this `validator.NewFunction`. I think this is much more compact and still understandable.

If you will keep old names I can share with you the article about package naming. I think `function-validator` is not the best name for the dir and `function_validator` for the package.
https://go.dev/blog/package-names

Reply:
done